org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BuildWorkspaceHandler.java

```diff
@@ -50,6 +52,18 @@ public BuildWorkspaceStatus buildWorkspace(boolean forceReBuild, IProgressMonito
 			if (monitor.isCanceled()) {
 				return BuildWorkspaceStatus.CANCELLED;
 			}
+			while (!BaseInitHandler.isWorkspaceInitialized()) {
+				try {
+					Thread.sleep(100);
+				} catch (InterruptedException e) {
+					// ignore and keep waiting
+				}
+			}
+			try {
+				Job.getJobManager().join(SyntaxInitHandler.JAVA_LS_INITIALIZATION_JOBS, null);
+			} catch (OperationCanceledException | InterruptedException e) {
+				logException(e.getMessage(), e);
+			}
```

Comment:
Does this support the usage of triggerFiles or is this some separate issue disovered ? I think we should avoid having requests from the client waiting on each other.

Reply:
Sometimes VS Code calls BuildWorkspaceHandler.buildWorkspace(boolean, IProgressMonitor) before initializing a workspace.
@rgrunber you have try the following:

- clean the Java LS workspace
- create a vscode-java vsix with and without this code
- install the vsix
- unzip [wrong-packagename](https://github.com/eclipse/eclipse.jdt.ls/files/6441935/wrong-packagename.zip)
```
cd wrong-packagename
code .
```

https://github.com/snjeza/vscode-test/raw/master/java-0.79.3.vsix includes this PR.

Reply:
@rgrunber @jdneo See https://github.com/redhat-developer/vscode-java/issues/1941#issuecomment-840014962